ipykernel/kernelbase.py

```diff
@@ -861,6 +862,13 @@ def raw_input(self, prompt=''):
         )
 
     def _input_request(self, prompt, ident, parent, password=False):
+        """Send an input request to the frontend and wait for the reply."""
+        # matplotlib needs to be imported after app.launch_new_instance()
+        try:
```

Comment:
Better to do `if 'matplotlib' in sys.modules: ...` as this will import `matplotlib` if it is installed, but not yet imported and it is not the quickest import.

I am not stoked about importing a private module here, I would do something like

```python
if 'matplotlib.pyplot' in sys.modules:
    def flush():
        import matplotlib.pyplot as plt
        if plt.get_fignums():
            plt.gcf().canvas.flush_events()
else:
    flush = None
```

Reply:
I imported that as this is the same import made here: 
https://github.com/ipython/ipykernel/blob/d283077435138f6bb5d77624e887a2552574d773/ipykernel/pylab/backend_inline.py#L11
and here in `plt.pause`:
https://github.com/matplotlib/matplotlib/blob/5e9c54e7c4feb296b387fadb56394ef67ac2927f/lib/matplotlib/pyplot.py#L307
I am not sure if there is a subtle difference between `Gcf.get_active().canvas.flush_events()` and `plt.gcf().canvas.flush_events()`?

Reply:
`Gcf` is not part of the public API (not withstanding the other usage in IPython).

Reply:
`plt.gcf().canvas.flush_events()` creates a new figure if no figures are open. Is there any way to avoid that?

Reply:
`plt.get_current_fig_manager()` has the same problem. It creates an active figure as a side effect.

Reply:
Sorrt I missed the `plt.get_fignums()`